addons/medical/functions/fnc_treatmentAdvanced_medicationLocal.sqf

```diff
@@ -73,8 +73,15 @@ if (alive _target) then {
 
 if (_painReduce > 0) then {
     // Reduce the pain level
-    _painSuppress = _target getvariable [QGVAR(painSuppress), 0];
-    _target setvariable [QGVAR(painSuppress), (_painSuppress + _painReduce) max 0];
+    if (GVAR(painIsOnlySuppressed)) then {
+        // Suppress the pain
+        _painSuppress = _target getvariable [QGVAR(painSuppress), 0];
+        _target setvariable [QGVAR(painSuppress), (_painSuppress + _painReduce) max 0];
+    } else {
```

Comment:
I would suggest to move the pain suppress above the if statement and do this instead:
```sqf
if (!GVAR(painIsOnlySuppressed)) then {
    _pain = _target getvariable [QGVAR(pain), 0];
    _target setvariable [QGVAR(pain), (_pain - _painReduce) max 0, true];
};
```
That has the benefit of still suppressing pain when it's added after morphine has been given.